src/neuroconv/utils/json_schema.py

```diff
@@ -70,104 +69,93 @@ def get_base_schema(
-                else:
-                    args_spec[param_name].update(additionalProperties=True)
-            elif hasattr(param.annotation, "__args__"):  # Annotation has __args__ if it was made by typing.Union
-                args = param.annotation.__args__
-                valid_args = [x.__name__ in annotation_json_type_map for x in args]
-                if not any(valid_args):
-                    raise ValueError(f"No valid arguments were found in the json type mapping for parameter {param}")
-                arg_types = [x for x in np.array(args)[valid_args]]
-                param_types = [annotation_json_type_map[x.__name__] for x in arg_types]
-                num_params = len(set(param_types))
-                conflict_message = (
-                    "Conflicting json parameter types were detected from the annotation! "
-                    f"{param.annotation.__args__} found."
-                )
-                # Normally cannot support Union[...] of multiple annotation types
-                if num_params > 2:
-                    raise ValueError(conflict_message)
-                # Special condition for Optional[...]
-                if num_params == 2 and args[1] is not type(None):  # noqa: E721
-                    raise ValueError(conflict_message)
-
-                # Guaranteed to only have a single index by this point
-                args_spec[param_name]["type"] = param_types[0]
-                if arg_types[0] == FilePathType:
-                    input_schema["properties"].update({param_name: dict(format="file")})
-                elif arg_types[0] == FolderPathType:
-                    input_schema["properties"].update({param_name: dict(format="directory")})
-            else:
-                arg = param.annotation
-                if arg.__name__ in annotation_json_type_map:
-                    args_spec[param_name]["type"] = annotation_json_type_map[arg.__name__]
-                else:
-                    raise ValueError(
-                        f"No valid arguments were found in the json type mapping '{arg}' for parameter {param}"
-                    )
-                if arg == FilePathType:
-                    input_schema["properties"].update({param_name: dict(format="file")})
-                if arg == FolderPathType:
-                    input_schema["properties"].update({param_name: dict(format="directory")})
-        else:
-            raise NotImplementedError(
-                f"The annotation type of '{param}' in function '{method}' is not implemented! "
-                "Please request it to be added at github.com/catalystneuro/nwb-conversion-tools/issues "
-                "or create the json-schema for this method manually."
+    for parameter_in_docstring in parsed_docstring.params:
+        if (
+            parameter_in_docstring.arg_name in json_schema["properties"]
+            and parameter_in_docstring.description is not None
```

Comment:
consider throwing a warning

Reply:
Added and tested

Reply:
Mmm I don't really like this. I think that warnings should be actionable. If someone pushed a function to the repo with a typo  and then someone else uses the functionthey will be getting warnings that they can't do nothing about. What use is this information to them? If we want the creator of the function to really add a correct docstring then maybe it should be an assertion or handled somewhere else I feel.

Reply:
I don't like to put that burden on the user and I don't see why we need to do it if we can inform the developer.

Reply:
Another thought

In theory, this method is only currently applied to core supported data interfaces in NeuroConv - do we trust ourselves enough to catch any such typos in `__init__` and `add_to_nwbfile` signatures+docstrings? If so then no worry

But actually then, when a **user** creates their own custom data interface with their own custom signatures and their own docstrings, then this becomes a useful warning in that case that they wouldn't have noticed without their own linting system

Reply:
Yeah, for that reason I would rather made an assertion. That way we tell the developer of the problem **but also** we don't allow him to be lazy and pass this standard error pollution to the end user.

Reply:
Can you open a subsequent PR elevating it to an error then? For discussion

Reply:
If you really prefer the warning then I think that I would rather have @bendichter break the vote here for assertion vs warning.

If he does not vote I will assume that he prefers a warning and we can let this aside.

I don't think it makes sense to open and issue, everything has been said here.